src/borg/archive.py

```diff
@@ -1387,21 +1389,29 @@ def process_item(self, archive, target, item):
 
     def process_chunks(self, archive, target, item):
         """Return new chunk ID list for 'item'."""
-        # TODO: support --compression-from
         if not self.recompress and not target.recreate_rechunkify:
             for chunk_id, size, csize in item.chunks:
                 self.cache.chunk_incref(chunk_id, target.stats)
             return item.chunks
         new_chunks = self.process_partial_chunks(target)
         chunk_iterator = self.create_chunk_iterator(archive, target, item)
         consume(chunk_iterator, len(new_chunks))
+        compress = self.compression_decider1.decide(item.path)
```

Comment:
I see. I also don't really see how to fix it, the information whether a given path was absolute or relative is thrown away for good.

```
borg create testrepo::1235 /home/mabe/Projekte/_oss/borg/src --debug
using builtin fallback logging configuration
29 self tests completed in 0.13 seconds
Reading files cache ...
/home/mabe/Projekte/_oss/borg/src/borg/__pycache__/upgrader.cpython-35.pyc -> compression none
```

```
borg create testrepo::1234 src --debug
using builtin fallback logging configuration
29 self tests completed in 0.14 seconds
Reading files cache ...
src/borg/chunker.cpython-35m-x86_64-linux-gnu.so -> compression none
Upgrading from v1 hints.159
Upgrade to v2 hints complete
src/borg/hashindex.c -> compression none
```